include/selene/ResourceHandler.h

```diff
@@ -2,21 +2,19 @@
 
 namespace sel {
 
-template<typename Signature>
-class CallOnce;
-
-template<typename... Args>
-class CallOnce<void(Args...)> {
-    std::function<void(Args...)> _f;
+template<typename Class, typename... Args>
+class CallOnceMemFun {
```

Comment:
This Abstraction should now be inlined directly in Selector. That would allow to get rid of a call to a member function through a pointer and should raise the chance for the compiler to actually inline.